server/src/main/java/org/opensearch/cluster/metadata/ResourceLimitGroup.java

```diff
@@ -0,0 +1,288 @@
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Class to define the ResourceLimitGroup schema
+ * {
+ *     "name": "analytics",
+ *     "resourceLimits": [
+ *          {
+ *              "resourceName": "jvm",
+ *              "value": 0.4
+ *          }
+ *     ]
+ * }
+ */
+@ExperimentalApi
+public class ResourceLimitGroup extends AbstractDiffable<ResourceLimitGroup> implements ToXContentObject {
+
+    public static final int MAX_CHARS_ALLOWED_IN_NAME = 50;
+    private final String name;
+    private final List<ResourceLimit> resourceLimits;
+    private final ResourceLimitGroupMode mode;
+
+    private static final List<String> ALLOWED_RESOURCES = List.of("jvm");
+
+    public static final ParseField NAME_FIELD = new ParseField("name");
+    public static final ParseField RESOURCE_LIMITS_FIELD = new ParseField("resourceLimits");
+    public static final ParseField MODE_FIELD = new ParseField("mode");
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<ResourceLimitGroup, Void> PARSER = new ConstructingObjectParser<>(
+        "ResourceLimitGroupParser",
+        args -> new ResourceLimitGroup((String) args[0], (List<ResourceLimit>) args[1], (String) args[2])
+    );
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), NAME_FIELD);
+        PARSER.declareObjectArray(
+            ConstructingObjectParser.constructorArg(),
+            (p, c) -> ResourceLimit.fromXContent(p),
+            RESOURCE_LIMITS_FIELD
+        );
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), MODE_FIELD);
+    }
+
+    public ResourceLimitGroup(final String name, final List<ResourceLimit> resourceLimits, final String modeName) {
+        Objects.requireNonNull(name, "ResourceLimitGroup.name can't be null");
+        Objects.requireNonNull(resourceLimits, "ResourceLimitGroup.resourceLimits can't be null");
+
+        if (name.length() > MAX_CHARS_ALLOWED_IN_NAME) {
+            throw new IllegalArgumentException("ResourceLimitGroup.name shouldn't be more than 50 chars long");
+        }
+
+        if (resourceLimits.isEmpty()) {
+            throw new IllegalArgumentException("ResourceLimitGroup.resourceLimits should at least have 1 resource limit");
+        }
+
+        this.name = name;
+        this.resourceLimits = resourceLimits;
+        this.mode = ResourceLimitGroupMode.fromName(modeName);
+    }
+
+    public ResourceLimitGroup(StreamInput in) throws IOException {
+        this(in.readString(), in.readList(ResourceLimit::new), in.readString());
+    }
+
+    /**
+     * Class to hold the system resource limits;
+     * sample Schema
+     *
+     */
+    @ExperimentalApi
+    public static class ResourceLimit implements Writeable, ToXContentObject {
+        private final String resourceName;
+        private final Double value;
```

Comment:
i think `limit` or `threshold` are better than `value` ?
`value` seems a little vague to me

Reply:
that is true in the context.